tests/unit_tests/audit_log/test_audit_query_handler.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from typing import Generic
+from unittest.mock import Mock
+
+from exasol.analytics.audit.audit_query_handler import (
+    AuditQueryHandler,
+    ParameterType,
+    ResultType,
+)
+from exasol.analytics.query_handler.query.select import (
+    DbOperationType,
+    ModifyQuery,
+    Query,
+    SelectQueryWithColumnDefinition,
+)
+from exasol.analytics.query_handler.query_handler import QueryHandler
+from exasol.analytics.query_handler.result import (
+    Continue,
+    Finish,
+)
+from exasol.analytics.schema import (
+    Column,
+    decimal_column,
+)
+from exasol.analytics.schema.schema_name import SchemaName
+from exasol.analytics.schema.table_name import TableName
+from exasol.analytics.schema.table_name_impl import TableNameImpl
+from tests.utils.audit_table_utils import (
+    QueryStringCriterion,
+    assert_queries_match,
+    create_insert_query,
+    expected_query,
+)
+
+
+@dataclass
+class MyParameterType(Generic[ParameterType]):
+    db_schema: str
+    table_name_prefix: str
+
+
+AUDIT_TABLE_NAME_PREFIX = "PPP"
+
+AUDIT_TABLE_NAME = TableNameImpl(
+    f"{AUDIT_TABLE_NAME_PREFIX}_AUDIT_LOG", SchemaName("SSS")
+)
+
+
+def create_audit_query_handler(
+    child: QueryHandler[ParameterType, ResultType],
+) -> AuditQueryHandler:
+    def schema_getter(parameter: MyParameterType) -> str:
+        return parameter.db_schema
+
+    def table_name_prefix_getter(parameter: MyParameterType) -> str:
+        return parameter.table_name_prefix
+
+    def additional_columns_provider(parameter: MyParameterType) -> list[Column]:
+        return [decimal_column("DDD", precision=9)]
+
+    parameter = MyParameterType(
+        db_schema="SSS",
+        table_name_prefix=AUDIT_TABLE_NAME_PREFIX,
+    )
+    return AuditQueryHandler(
+        parameter=parameter,
+        context=Mock(),
+        query_handler_factory=Mock(return_value=child),
+        schema_getter=schema_getter,
+        table_name_prefix_getter=table_name_prefix_getter,
+        additional_columns_provider=additional_columns_provider,
+    )
+
+
+def test_constructor() -> None:
+    child = Mock()
+    testee = create_audit_query_handler(child)
+    audit_table = testee._audit_table
+    assert audit_table.name == AUDIT_TABLE_NAME
+    assert '"DDD" DECIMAL(9,0)' in [c.for_create for c in audit_table.columns]
+    assert child == testee._child.query_handler
+
+
+def test_start_finish():
```

Comment:
```suggestion
def test_start_finish_without_audit_query():
```

Reply:
Additional to these tests, you should also add tests with PythonQueryHandlerRunner, because it simulates better the restriction of the system, aka the column definition match for the return_query of the Continue.